### PR TITLE
Update to OpenTelemetryv0.14.0

### PR DESCRIPTION
**Pull request by kowshikn**
*Issue #151 *

*Description of changes:*

* Merging the upgrade to latest OpenTelemetry Span spec. Kibana has pushed the [change](https://github.com/opendistro-for-elasticsearch/trace-analytics/pull/3) and will be available part of ODFE Kibana 7.10.0 release scheduled on 12/10.
* In the sample use the jar from kibana team which uses ```status.code==2```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kowshikn** [comment]
Analytics service is not publishing trace data. Will investigate this one.

**Review by yadavcbala — APPROVED** [review]

File: examples/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FROM gradle:jdk14 AS builder
 COPY . /home/gradle/src
 WORKDIR /home/gradle/src/situp-core
-RUN gradle jar --daemon
+RUN gradle cleanjar --daemon
```
Comment (chenqi0805) [inline examples/Dockerfile:4]: This does not work on my laptop.

Comment (kowshikn) [reply examples/Dockerfile:4]: My bad, this was a typo. Fixing in the next commit. `RUN gradle clean jar --daemon`

**Comment by kowshikn** [comment]
> Analytics service is not publishing trace data. Will investigate this one.

`[opentelemetry.auto.trace 2020-11-18 19:01:36:800 +0000] [grpc-default-executor-0] WARN io.opentelemetry.exporter.otlp.OtlpGrpcSpanExporter - Failed to export spans. Error message: UNAVAILABLE: io exception
 `
Looks like Analytics-service is having issues after version upgrade.

**Comment by chenqi0805** [comment]
FTR offline discussion. There is manual introduction of random error:
```
@PostMapping("/logs")
    public String save(@RequestBody Logging logging) throws IOException {
        //TODO send logs to another system.
        return "Dummy";
        // 1% of the time will throw exception
        if (RANDOM.nextInt(100) == 0) {
            throw new IOException("Dummy Exception");
        } else {
            return "Dummy";
        }
    }
```

**Review by chenqi0805 — APPROVED** [review]
Confirmed that by reverting the random error introduced in analytics-service, there will be no status code = 2 related to /logs. So nothing blocking AFAICT.

**Review by chenqi0805 — APPROVED** [review]

**Pull request merged**